File: CS/MessageBoxHelper/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace MessageBoxHelper {
    public partial class MainForm : Form {
        public MainForm() {
            InitializeComponent();
            myXtraMessageArgsBindingSource.DataSource = new MyXtraMessageArgs();
            AutoCloseCheckEdit.Enabled = false;
            ShowCountdownCheckEdit.Enabled = false;
            ButtonsComboBox.Properties.Items.AddEnum(typeof(MessageBoxButtons));
            AutoCloseResultComboBoxEdit.Enabled = AutoCloseCheckEdit.Checked;
            foreach (string iconName in Enum.GetNames(typeof(MessageBoxIcon)))
                IconComboBox.Properties.Items.Add(iconName);
        }

        void UpdateAutoClose() {
            AutoCloseResultComboBoxEdit.Properties.Items.Clear();
            AutoCloseResultComboBoxEdit.EditValue = null;
            if (!AutoCloseResultComboBoxEdit.Enabled)
                return;
            if (ButtonsComboBox.EditValue != null) {
                DialogResult[] buttons = MyXtraMessageBox.GetDialogResultsFromButtons((MessageBoxButtons)ButtonsComboBox.EditValue);
                AutoCloseResultComboBoxEdit.Properties.Items.AddRange(buttons);
            } else
                AutoCloseResultComboBoxEdit.Properties.Items.Add(DialogResult.OK);
        }

        private void ButtonsComboBox_EditValueChanged(object sender, EventArgs e) {
            UpdateAutoClose();
        }

        private void btnShow_Click(object sender, EventArgs e) {
            DialogResult result = MyXtraMessageBox.Show(myXtraMessageArgsBindingSource.DataSource as MyXtraMessageArgs);
            txtResult.Text = result.ToString();
            myXtraMessageArgsBindingSource.ResetBindings(false);
        }

        private void AutoCloseCheckEdit_CheckedChanged(object sender, EventArgs e) {
            AutoCloseResultComboBoxEdit.Enabled = AutoCloseCheckEdit.Checked;
            UpdateAutoClose();
        }

        private void TimeoutSpinEdit_EditValueChanged(object sender, EventArgs e) {
            if (Convert.ToInt32(TimeoutSpinEdit.EditValue) <= 0) {
                AutoCloseCheckEdit.Checked = false;
                AutoCloseCheckEdit.Enabled = false;
                ShowCountdownCheckEdit.Checked = false;
                ShowCountdownCheckEdit.Enabled = false;
            } else {
                AutoCloseCheckEdit.Enabled = true;
                ShowCountdownCheckEdit.Enabled = true;
            }
        }

        private void ShowMessageNextTimeCheckEdit_EditValueChanged(object sender, EventArgs e) {

        }
    }
}

File: VB/MessageBoxHelper/MainForm.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace MessageBoxHelper
    Partial Public Class MainForm
        Inherits Form

        Public Sub New()
            InitializeComponent()
            myXtraMessageArgsBindingSource.DataSource = New MyXtraMessageArgs()
            AutoCloseCheckEdit.Enabled = False
            ShowCountdownCheckEdit.Enabled = False
            ButtonsComboBox.Properties.Items.AddEnum(GetType(MessageBoxButtons))
            AutoCloseResultComboBoxEdit.Enabled = AutoCloseCheckEdit.Checked
            For Each iconName As String In System.Enum.GetNames(GetType(MessageBoxIcon))
                IconComboBox.Properties.Items.Add(iconName)
            Next iconName
        End Sub

        Private Sub UpdateAutoClose()
            AutoCloseResultComboBoxEdit.Properties.Items.Clear()
            AutoCloseResultComboBoxEdit.EditValue = Nothing
            If Not AutoCloseResultComboBoxEdit.Enabled Then
                Return
            End If
            If ButtonsComboBox.EditValue IsNot Nothing Then
                Dim buttons() As DialogResult = MyXtraMessageBox.GetDialogResultsFromButtons(CType(ButtonsComboBox.EditValue, MessageBoxButtons))
                AutoCloseResultComboBoxEdit.Properties.Items.AddRange(buttons)
            Else
                AutoCloseResultComboBoxEdit.Properties.Items.Add(System.Windows.Forms.DialogResult.OK)
            End If
        End Sub

        Private Sub ButtonsComboBox_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles ButtonsComboBox.EditValueChanged
            UpdateAutoClose()
        End Sub

        Private Sub btnShow_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnShow.Click
            Dim result As DialogResult = MyXtraMessageBox.Show(TryCast(myXtraMessageArgsBindingSource.DataSource, MyXtraMessageArgs))
            txtResult.Text = result.ToString()
            myXtraMessageArgsBindingSource.ResetBindings(False)
        End Sub

        Private Sub AutoCloseCheckEdit_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles AutoCloseCheckEdit.CheckedChanged
            AutoCloseResultComboBoxEdit.Enabled = AutoCloseCheckEdit.Checked
            UpdateAutoClose()
        End Sub

        Private Sub TimeoutSpinEdit_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles TimeoutSpinEdit.EditValueChanged
            If Convert.ToInt32(TimeoutSpinEdit.EditValue) <= 0 Then
                AutoCloseCheckEdit.Checked = False
                AutoCloseCheckEdit.Enabled = False
                ShowCountdownCheckEdit.Checked = False
                ShowCountdownCheckEdit.Enabled = False
            Else
                AutoCloseCheckEdit.Enabled = True
                ShowCountdownCheckEdit.Enabled = True
            End If
        End Sub

        Private Sub ShowMessageNextTimeCheckEdit_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles ShowMessageNextTimeCheckEdit.EditValueChanged

        End Sub
    End Class
End Namespace
